sound/hda/ext/hdac_ext_stream.c

```diff
@@ -296,12 +296,13 @@ hdac_ext_host_stream_assign(struct hdac_bus *bus,
 		struct hdac_ext_stream *hstream = container_of(stream,
 						struct hdac_ext_stream,
 						hstream);
-		if (stream->direction != substream->stream)
+		if (stream->direction != direction)
 			continue;
 
 		if (!stream->opened) {
 			if (!hstream->decoupled)
-				snd_hdac_ext_stream_decouple(bus, hstream, true);
+				snd_hdac_ext_stream_decouple(bus,
+						hstream, true);
```

Comment:
leave as is, it's more readable on one line

Reply:
Will generate 80line warning if I'm not mistaken. I'm fine with keeping it one-liner though.

Reply:
Fixed in v2.